src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -263,17 +264,59 @@ CHIP_ERROR DiscoveryImplPlatform::Advertise(const OperationalAdvertisingParamete
     mOperationalAdvertisingParams = params;
     // TODO: There may be multilple device/fabrid ids after multi-admin.
 
-    ReturnErrorOnFailure(SetupHostname(params.GetMac()));
+    TextEntry crmpRetryIntervalEntries[2];
+    size_t textEntrySize = 0;
+    char crmpRetryIntervalIdleBuf[8];
+    char crmpRetryIntervalActiveBuf[8];
+    uint32_t crmpRetryIntervalIdle, crmpRetryIntervalActive;
+    params.GetCRMPRetryIntervals(crmpRetryIntervalIdle, crmpRetryIntervalActive);
+
+    // TODO: Issue #5833 - CRMP retry intervals should be updated on the poll period value
+    // change or device type change.
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    if (chip::DeviceLayer::ConnectivityMgr().GetThreadDeviceType() ==
+        chip::DeviceLayer::ConnectivityManager::kThreadDeviceType_SleepyEndDevice)
+    {
+        uint32_t sedPollPeriod;
+        ReturnErrorOnFailure(chip::DeviceLayer::ThreadStackMgr().GetPollPeriod(sedPollPeriod));
+        // Increment default CRMP retry intervals by SED poll period to be on the safe side
+        // and avoid unnecessary retransmissions.
+        crmpRetryIntervalIdle += sedPollPeriod;
+        crmpRetryIntervalActive += sedPollPeriod;
+    }
+#endif
 
+    if (crmpRetryIntervalIdle > 3600000)
```

Comment:
nit: any chance to make the number 3600000.. constant - it repeats few times below?